### PR TITLE
Add lora adapter management API

### PR DESCRIPTION
**Pull request by brosoul**
## Pull Request Description
Add API interface to manage lora adapter
- `/lora_adapter/load` to load lora adapter
- `/lora_adapter/unload` to unload lora adapter

At the same time, it provides a base class `InferenceEngine` interface for abstract engine side interaction.

## Related Issues
Resolves: #186 

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

File: python/aibrix/aibrix/envs.py [hunk 1]
```diff
@@ -87,3 +87,10 @@ def _parse_int_or_none(value: Optional[str]) -> Optional[int]:
 
 # Runtime Metric config
 PROMETHEUS_MULTIPROC_DIR = os.getenv("PROMETHEUS_MULTIPROC_DIR", "/tmp/aibrix/metrics/")
+
+# Inference Engine Config
+INFERENCE_ENGINE = os.getenv("INFERENCE_ENGINE", "vllm")
```
Comment (brosoul) [inline python/aibrix/aibrix/envs.py:89]: Should the environment variables here be reused with `METRIC_SCRAPE_xxx`? 
My viewpoint is that although some values overlap, it is still necessary to split them into multiple variables for readability. Looking forward to your advice @Jeffwan

File: python/aibrix/aibrix/envs.py [hunk 2]
```diff
@@ -87,3 +87,10 @@ def _parse_int_or_none(value: Optional[str]) -> Optional[int]:
...
+INFERENCE_ENGINE_VERSION = os.getenv("INFERENCE_ENGINE_VERSION", "0.6.1")
```
Comment (Jeffwan) [inline python/aibrix/aibrix/envs.py:90]: what's the version for?

File: python/aibrix/aibrix/envs.py (hunk 1 above)
Comment (Jeffwan) [reply python/aibrix/aibrix/envs.py:89]: I didn't get this part. `METRIC_SCRAPE_xxx `, the xxx is the original metric name? Why do you leave the comment at the inference engine config here? any correlation?

File: python/aibrix/app.py [hunk 3]
```diff
@@ -61,9 +74,37 @@ def mount_metrics(app: FastAPI):
...
 
+def init_app_state(state: State) -> None:
+    state.inference_engine = get_inference_engine(
+        envs.INFERENCE_ENGINE,
+        envs.INFERENCE_ENGINE_VERSION,
+        envs.INFERENCE_ENGINE_ENDPOINT,
+    )
+
+
+@router.post("/lora_adapter/load")
```
Comment (Jeffwan) [inline python/aibrix/app.py]: let's add api version here? like `/v1/lora_adapter/load`

File: python/aibrix/aibrix/envs.py (hunk 2 above)
Comment (Jeffwan) [reply python/aibrix/aibrix/envs.py:90]: just check the following code, notice it's for feature gate checking. make sense.

File: python/aibrix/aibrix/envs.py (hunk 1 above)
Comment (brosoul) [reply python/aibrix/aibrix/envs.py:89]: For example, the values of `METRIC_SCRAPE_ENGINE` and `INFERENCE_ENGINE` are the same. `INFERENCE_ENGINE_ENDPOINT` is also related to the combination of `METRIC_SCRAPE_HOST` and `METRIC_SCRAPE_PORT`.

File: python/aibrix/app.py (hunk 3 above)
Comment (brosoul) [reply python/aibrix/app.py]: done

File: python/aibrix/aibrix/envs.py (hunk 1 above)
Comment (Jeffwan) [reply python/aibrix/aibrix/envs.py:89]: this comment is discussed offline. It's better to consolidate the envs and have a consistent variable for multiple purpose.

**Pull request merged**